rllib/examples/models/parametric_actions_model.py

```diff
@@ -101,8 +101,9 @@ def forward(self, input_dict, state, seq_lens):
         # Mask out invalid actions (use -inf to tag invalid).
         # These are then recognized by the EpsilonGreedy exploration component
         # as invalid actions that are not to be chosen.
-        inf_mask = torch.clamp(
-            torch.log(action_mask), -float("inf"), float("inf"))
+        inf_mask = torch.clamp(torch.log(action_mask), -1e15, 1e15)
+        inf_mask = inf_mask.to(self.device)
```

Comment:
This may not be the culprit causing the device issue.  All inputs should've been transferred to the device already.

Reply:
@sven1977 any idea why we see the device error?

Reply:
I'm checking ...

Reply:
I think this is completely unrelated. The loss seems to be wrong, which is why we are getting NaNs (if we don't run on a GPU) and then the dist.sample() complains. When running on GPU, this same error gets caught earlier (through some CUDA gradient assert?). I can reproduce this now and try to come up with a fix.

Reply:
I think the copy to GPU is fine here. `inf_mask` is created on the CPU (from `action_mask`, which is as well) and must be copied to the GPU then.

However, the move from inf to e-15 breaks epsilon greedy, which also uses -inf/inf as masking. We need to make more changes to the EpsilonGreedy Exploration class to fix these.